### PR TITLE
groonga: add msgpack-c as default dependency

### LINKED ISSUE #272202: pgroonga's groonga not built with msgpack
### Describe the bug
Failed to load db with the following setup:

`flake.nix`
```nix
{
  description = "pgroonga-msgpack-issue";
  inputs = {
    nixpkgs.url = "github:NixOS/nixpkgs/nixpkgs-unstable";
    flake-utils.url = "github:numtide/flake-utils";
  };

  outputs = { self, nixpkgs, flake-utils }:
    flake-utils.lib.eachDefaultSystem (system: {
      devShell = let
        pkgs = import nixpkgs { inherit system; };
        postgres = pkgs.postgresql_15.withPackages (p: [p.pgroonga]);
        PRJ_ROOT = builtins.getEnv "PWD";
        pg-setup = pkgs.writeShellScriptBin "pg-setup" ''
          set -euo pipefail
          export PATH=${postgres}/bin:${pkgs.coreutils}/bin

          # exist if dir already exists
          [[ ! -d "$PGDATA" ]] || exit 0

          echo "initdb..."
          initdb --encoding=UTF8 --no-locale --no-instructions -U ''${PGUSER:-postgres}
          echo "initdb done."

          cat >> "$PGDATA/postgresql.conf" <<EOF
            listen_addresses = '''
            unix_socket_directories = '$PGHOST'
            log_statement = 'all'
          EOF
        '';
        pg-start = pkgs.writeShellScriptBin "pg-start" ''
          set -euo pipefail
          ${pg-setup}/bin/pg-setup
          exec ${postgres}/bin/postgres
        '';
        pg-load = pkgs.writeShellScriptBin "pg-load" ''
          export PATH=${postgres}/bin

          psql < ''${1:-$PRJ_ROOT/db.sql}
        '';
      in pkgs.mkShell {
        buildInputs = [ postgres pg-setup pg-start pg-load ];

        shellHook = ''
          export PRJ_ROOT="${PRJ_ROOT}";
          export PGDATA="${PRJ_ROOT}/.data/postgres";
          export PGHOST="$PGDATA";
          export PGUSER="postgres";

          pg-setup
        '';
      };
    });
}
```
`db.sql`
```pgsql
CREATE EXTENSION IF NOT EXISTS pgroonga;

CREATE TABLE IF NOT EXISTS normalizations (
  target text,
  normalized text
);

CREATE INDEX IF NOT EXISTS pgrn_normalizations_index ON normalizations
 USING pgroonga (target pgroonga_text_term_search_ops_v2,
                 normalized);

INSERT INTO normalizations VALUES ('o', '0');
INSERT INTO normalizations VALUES ('ss', '55');

CREATE TABLE IF NOT EXISTS memos (
  id integer,
  content text
);

CREATE INDEX IF NOT EXISTS pgroonga_memos_index
          ON memos
       USING pgroonga (content)
        WITH (normalizers='
                NormalizerNFKC130,
                NormalizerTable(
                  "normalized", "${table:pgrn_normalizations_index}.normalized",
                  "target", "target"
                )
             ');

INSERT INTO memos VALUES (1, '0123455');
INSERT INTO memos VALUES (2, 'o1234ss');

SELECT * FROM memos WHERE content &@~ 'o123455';
```

This is due to `pgroonga` requires `groonga` with `msgpack`.

### Steps To Reproduce
1. nix develop . --impure
2. pg-start &
3. pg-load

### Expected behavior
Load db without error.
```console
bash-5.2$ pg-load
CREATE EXTENSION
CREATE TABLE
CREATE INDEX
INSERT 0 1
INSERT 0 1
CREATE TABLE
CREATE INDEX
INSERT 0 1
INSERT 0 1
 id | content 
----+---------
  1 | 0123455
  2 | o1234ss
(2 rows)
```

### Additional context
pg-load result:
```console
bash-5.2$ pg-load
CREATE EXTENSION
CREATE TABLE
CREATE INDEX
INSERT 0 1
INSERT 0 1
CREATE TABLE
ERROR:  pgroonga: [option][normalizers][validate] invalid normalizers: <
                NormalizerNFKC130,
                NormalizerTable(
                  "normalized", "${table:pgrn_normalizations_index}.normalized",
                  "target", "target"
                )
             >: [options] MessagePack is required
INSERT 0 1
INSERT 0 1
 id | content 
----+---------
(0 rows)
```

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"aarch64-darwin"`
 - host os: `Darwin 23.1.0, macOS 14.1.1`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.18.1`
 - channels(root): `"nixpkgs"`
 - channels(eth3lbert): `""`
 - nixpkgs: `/nix/store/vxyz6n22g6dy11cnr8d7a2017766rfqi-source`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by eth3lbert**
pgroonga requires groonga with msgpack #272202.
Should close #272202.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by eth3lbert** [comment]
Result of `nixpkgs-review pr 272203` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>10 packages built:</summary>
  <ul>
    <li>postgresql12JitPackages.pgroonga</li>
    <li>postgresql12Packages.pgroonga</li>
    <li>postgresql13JitPackages.pgroonga</li>
    <li>postgresql13Packages.pgroonga</li>
    <li>postgresql14JitPackages.pgroonga</li>
    <li>postgresql14Packages.pgroonga</li>
    <li>postgresqlJitPackages.pgroonga (postgresql15JitPackages.pgroonga)</li>
    <li>postgresql15Packages.pgroonga</li>
    <li>postgresql16JitPackages.pgroonga</li>
    <li>postgresql16Packages.pgroonga</li>
  </ul>
</details>

File: pkgs/servers/sql/postgresql/ext/pgroonga.nix [hunk 1]
```diff
@@ -10,7 +10,17 @@ stdenv.mkDerivation rec {
   };
 
   nativeBuildInputs = [ pkg-config ];
-  buildInputs = [ postgresql msgpack-c groonga ];
+  buildInputs =
+    let
+      groonga-with-msgpack = groonga.overrideAttrs (oa: {
```
Comment (NickCao) [inline pkgs/servers/sql/postgresql/ext/pgroonga.nix:15]: Or should it be `groonga.override { suggestSupport = true; }`?

Comment (eth3lbert) [reply pkgs/servers/sql/postgresql/ext/pgroonga.nix:15]: `pgroonga` doesn't seem to use the suggest feature from `groonga`, would you recommend that we still use `suggestSupport` flag to build `groonga`?

Comment (NickCao) [reply pkgs/servers/sql/postgresql/ext/pgroonga.nix:15]: According to the upstream documentation: https://groonga.org/docs/install/debian.html#build-from-source, msgpack should be added as a dependency by default. So another choice would be to include `msgpack` (or `msgpack-c` if supported) in `buildInputs` unconditionally.

Comment (eth3lbert) [reply pkgs/servers/sql/postgresql/ext/pgroonga.nix:15]: I second this, It also a default dependency according to groonga's [brew formula](https://formulae.brew.sh/formula/groonga).

Comment (eth3lbert) [reply pkgs/servers/sql/postgresql/ext/pgroonga.nix]: Have added msgpack-c as default dependency, thanks for the review.

**Comment by eth3lbert** [comment]
Result of `nixpkgs-review` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>11 packages built:</summary>
  <ul>
    <li>groonga</li>
    <li>postgresql12JitPackages.pgroonga</li>
    <li>postgresql12Packages.pgroonga</li>
    <li>postgresql13JitPackages.pgroonga</li>
    <li>postgresql13Packages.pgroonga</li>
    <li>postgresql14JitPackages.pgroonga</li>
    <li>postgresql14Packages.pgroonga</li>
    <li>postgresqlJitPackages.pgroonga (postgresql15JitPackages.pgroonga)</li>
    <li>postgresql15Packages.pgroonga</li>
    <li>postgresql16JitPackages.pgroonga</li>
    <li>postgresql16Packages.pgroonga</li>
  </ul>
</details>

**Review by NickCao — APPROVED** [review]
diff LGTM, checked build log, `msgpack-c` is picked up.

Result of `nixpkgs-review pr 272203` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>11 packages built:</summary>
  <ul>
    <li>groonga</li>
    <li>postgresql12JitPackages.pgroonga</li>
    <li>postgresql12Packages.pgroonga</li>
    <li>postgresql13JitPackages.pgroonga</li>
    <li>postgresql13Packages.pgroonga</li>
    <li>postgresql14JitPackages.pgroonga</li>
    <li>postgresql14Packages.pgroonga</li>
    <li>postgresqlJitPackages.pgroonga (postgresql15JitPackages.pgroonga)</li>
    <li>postgresqlPackages.pgroonga (postgresql15Packages.pgroonga)</li>
    <li>postgresql16JitPackages.pgroonga</li>
    <li>postgresql16Packages.pgroonga</li>
  </ul>
</details>

cc @ericsagnes

**Pull request merged**